crypto/src/hash.rs

```diff
@@ -36,11 +36,11 @@ ffi::ffi_item! {
     )]
     #[repr(transparent)]
     #[serde(transparent)]
-    // TODO: use #[ffi_type(unsafe {robust})] instead
-    #[ffi_type(opaque)]
+    // SAFETY: Hash has no trap representations in [u8; LENGTH]
+    #[ffi_type(unsafe {robust})]
     #[display(fmt = "{}", "hex::encode(_0)")]
     #[debug(fmt = "{{ Hash({}) }}", "hex::encode(_0)")]
-    pub struct Hash([u8; Self::LENGTH]);
+    pub struct Hash([u8; Hash::LENGTH]);
```

Comment:
it would be nice if we could use zero hash for trap representation but we allow all zero hash as a signal

Reply:
yes, though it's not present yet, any method on `SignatureOf` that takes `HashOf`, etc.

Reply:
> any method on `SignatureOf` that takes `HashOf`, etc.

But this method takes `HashOf<T>` not `Option<HashOf<T>>`, or i miss something?

Reply:
yes, but `HashOf` will be transmuted into `Hash` which will be transmuted into `[u8; u32]`. The function will end up ingesting `[u8; u32]`

Reply:
and what benefits it would give us if all zeroes would be considered niche value? 
i thought we use it only in combination with `Option`

Reply:
you are correct, `Option` is required, this wasn't a good example